### PR TITLE
GHA CI and PHP 8

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

This pull add the GHA CI setup, removing travis config.
It also refactors tests similar to #5 with ref to #4 but that pull looks stuck

The reason for the pull is to hopefully get it merged so that I can tackle adding Psalm and possibly update the coding standard to ^2

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -28,9 +28,9 @@
         }
     },
     "require": {
-        "php": "^7.1",
+        "php": "^7.3||8.0",
```
Comment (froschdesign) [inline composer.json]: PHP does not follow semantic versioning, therefore:
```suggestion
        "php": "^7.3 || ~8.0.0",
```

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    needs: [matrix]
+    runs-on: ${{ matrix.operatingSystem }}
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJSON(needs.matrix.outputs.matrix) }}
+    steps:
+      - name: ${{ matrix.name }}
+        uses: laminas/laminas-continuous-integration-action@v1
+        with:
+          job: ${{ matrix.job }}
```
Comment (froschdesign) [inline .github/workflows/continuous-integration.yml]: This small error should be fixed. Thanks!

**Comment by gsteel** [comment]
Thank you for the review @froschdesign - I have also rebased on 3.x

File: composer.json [hunk 3]
```diff
@@ -28,9 +28,9 @@
         }
     },
     "require": {
-        "php": "^7.1",
+        "php": "^7.3 || ~8.0",
```
Comment (froschdesign) [inline composer.json]: Be careful here:

> The ~ operator is best explained by example: `~1.2` is equivalent to `>=1.2 <2.0.0`, while `~1.2.3` is equivalent to `>=1.2.3 <1.3.0`.

https://getcomposer.org/doc/articles/versions.md#tilde-version-range-

```suggestion
        "php": "^7.3 || ~8.0.0",
```

**Comment by gsteel** [comment]
Leaving a polite nag for review from perhaps @froschdesign or @weierophinney … Sorry … I don't know who the maintainers are for this package 😬

**Pull request merged**